src/audio/volume/volume.c

```diff
@@ -1181,6 +1179,74 @@ static vol_zc_func vol_get_zc_function(struct comp_dev *dev)
 	return NULL;
 }
 
+/**
+ * \brief Set volume frames alignment limit.
+ * \param[in,out] source Source structure.
+ * \param[in,out] sink Sink structure.
+ */
+#if XCHAL_HAVE_HIFI3 || XCHAL_HAVE_HIFI4
+
+/**
+ * in HiFi3 or HiFi4, we use multi-way operation intrinsics, those operations
+ * require 8-byte aligned, and 5.1 format SSE audio asked for 16-byte aligned,
+ * so we should aligned with 8 or 16 bytes. Since there is at most
+ * SOF_IPC_MAX_CHANNELS channels, the following 2 arrays are the restriction
+ * tables of 16 and 32 bit per sample.
+ */
+
+const struct processing_alignment volume_align_16bit[SOF_IPC_MAX_CHANNELS] = {
+	/** to meet the requirement of x-byte aligned,the frame_size * frame_align
+	 * should be multiple of x
+	 */
+	/* channel_num, alignment_shift_idx */
+	{1, 2}, /*8-byte aligned, frame size: 2 bytes, frame_align: 4 frames*/
```

Comment:
looks better, but the shift index should be calculated using a macro that takes the frame size and byte or frame alignment as arguments. A developer is going to be thinking about the frame size and byte alignment when developing the code and may not understand the internal shifts we do to optimize.